logger.go

```diff
@@ -69,7 +72,49 @@ func (l *MicroLogger) LogCtx(ctx context.Context, keyVals ...interface{}) error
 }
 
 func (l *MicroLogger) With(keyVals ...interface{}) Logger {
+	keyVals = l.processStack(keyVals)
 	return &MicroLogger{
 		logger: kitlog.With(l.logger, keyVals...),
 	}
 }
+
+func (l *MicroLogger) processStack(keyVals []interface{}) []interface{} {
+	for i := 1; i < len(keyVals); i += 2 {
+		k := keyVals[i-1]
+		v := keyVals[i]
```

Comment:
Does the code panic when an uneven number of elements is provided? I think the logger we use under the hood does but it would be good to prevent our code from doing something even more weird. Should we just return what we get when the number of key values is uneven? What are your thoughts on this?

Reply:
No because if `i >= len(keyVals)` the loop stops. The loop starts from 1 and adds 2 so it ignores the last element if len(keyVals) yields uneven value. When number of key is uneven then `(MISSING)` is printed. We should probably add a test case for that.

Reply:
Test case added. You can see `(MISSING)` here: https://github.com/giantswarm/micrologger/pull/26/files#diff-21a7c5b556072d55519560ccdfea8a05

Nothing panics.